protostar/cli/protostar_arg_type.py

```diff
@@ -18,27 +19,35 @@
     "address",
     "class_hash",
     "block_explorer",
+    "contract_source_identifier",
 ]
 
 ProtostarArgTypeName = Union[CustomProtostarArgTypeName, ArgTypeName]
 
 # pylint: disable=too-many-return-statements
-def map_protostar_type_name_to_parser(
-    argument_type: ProtostarArgTypeName,
-) -> Callable[[str], Any]:
-    if argument_type == "felt":
-        return parse_felt_arg_type
-    if argument_type == "fee":
-        return parse_fee_arg_type
-    if argument_type == "address":
-        return parse_hex_or_decimal
-    if argument_type == "class_hash":
-        return parse_hex_or_decimal
-    if argument_type == "wei":
-        return parse_wei_arg_type
-    if argument_type == "block_explorer":
-        return parse_block_explorer_type
-    return map_type_name_to_parser(argument_type)
+def create_map_protostar_type_name_to_parser(
```

Comment:
Please stick to factories or use a partial

Reply:
Partial doesn't preserve types. Changed it to factory.